### PR TITLE
v1.2.1 changelogs

### LINKED ISSUE #255: CWL v1.2.1 Changelog
Please report changes here, thanks

### PR DESCRIPTION
**Pull request by mr-c**
Fixes  #255

Direct links to previews of the 3 changelogs:
- https://deploy-preview-262--cwl-v1-2-dev.netlify.app/commandlinetool#New_Mandatory_Conformance_Tests_for_v1.2.1
- https://deploy-preview-262--cwl-v1-2-dev.netlify.app/workflow#Changelog_for_v1.2.1
- https://deploy-preview-262--cwl-v1-2-dev.netlify.app/schemasalad#Changelog_for_v1.2.1

### CODE REVIEW

File: CommandLineTool.yml [hunk 1]
```diff
@@ -37,10 +37,224 @@ $graph:
...
+        the fields of the record.
+      * `staging-basename`: Use of an `ExpressionTool` to change `basename` of
+        file, then correctly staging the file using the new name.
+      * `js-input-record`:  A test case for JavaScript with an input record.
+      * `very_big_and_very_floats`: Confirm that very big and very small
+        numbers are represented on the command line using decimals, not
+        scientific notation.
+
+      #### `MultipleInputFeatureRequirement`
+      * `multiple-input-feature-requirement`: MultipleInputFeatureRequirement on
```
Comment (kinow) [inline CommandLineTool.yml:248]: Above `InlineJavascriptRequirement` is formatted, but here `MultipleInputFeatureRequirement` is not. Probably doesn't matter :+1:

File: CommandLineTool.yml [hunk 2]
```diff
@@ -37,10 +37,224 @@ $graph:
...
+        objects automatically for the user.
+      * The schema for the `hints` field of `CommandLineTool` has been expanded
+        from: `Any[]?` to `["null", { type: array, items: [ ProcessRequirement, Any] } ]`.
+        This allows autogenerated CWL parsers to deserialize any of the standard
+        CWL hints instead of forcing the users of those parsers to convert the
+        unserialized hints to normal objects themselves.
+
+      ### Updated Conformance Tests for v1.2.1
+      * Conformance tests are now referred to by their textual identifiers (`id`). Previously
+        this was the `label` field. Tests without a `label`/`id` have been give
```
Comment (kinow) [inline CommandLineTool.yml:155]: s/give/given

File: Workflow.yml [hunk 3]
```diff
@@ -34,6 +34,170 @@ $graph:
...
+        are never type-checked due to a long-standing bug in the CWL reference
+        implementation. This has been made explicit along with the plan to fix
+        this oversight in CWL v1.3.
+      * The purpose and valid circumstances for using `Workflow.id`, `ExpressionTool.id`,
+        or `Operation.id` has been made more explicit: It is a unique identifier for that Process;
+        Only useful for when those are in a `$graph`. This `id` value should
+        not be exposed to users in graphical or terminal user interfaces.
+
+      ### Clarifications to the schema in CWL v1.2.1 to aid autogenerated libraries
+      Many CWL parsing/generating libraries are autogenerated from the official schema
```
[the remark this answers is not in the document]
Comment (mr-c) [reply Workflow.yml:95]: We don't render the website using GitHub; thanks for checking.

https://deploy-preview-262--cwl-v1-2-dev.netlify.app/workflow#Clarifications_to_the_schema_in_CWL_v1.2.1_to_aid_autogenerated_libraries

File: Workflow.yml [hunk 4]
```diff
@@ -34,6 +34,170 @@ $graph:
...
+
+      This specification represents the latest development draft from the
+      CWL project. There are no new features nor behavior changes in CWL v1.2.1
+      as compared to CWL v1.2. v1.2.1 fixes only typos, adds clarifications,
+      and adds additional conformance tests. Some changes to the schema defining
+      CWL have been changed to aid the auto-generation of libraries for the reading
+      and writing of CWL documents.
+
+      Do not write `cwlVersion: v1.2.1`, that is not a valid version of the CWL
+      syntax.
```
[the remark this answers is not in the document]
Comment (mr-c) [reply Workflow.yml]: Fixed in 0186cc8

**Review by kinow — APPROVED** [review]
Thanks!

**Pull request merged**